src/notification.c

```diff
@@ -296,7 +287,16 @@ int notification_init(notification *n, int id)
         assert(n != NULL);
 
         //Prevent undefined behaviour by initialising required fields
-        notification_init_defaults(n);
+        if(n->appname == NULL) n->appname = g_strdup("unknown");
+        if(n->summary == NULL) n->summary = g_strdup("");
+        if(n->body == NULL) n->body = g_strdup("");
+        if(n->category == NULL) n->category = g_strdup("");
+        n->redisplayed = false;
+        n->first_render = true;
+        n->script = NULL;
```

Comment:
Nitpick but not sure these are needed, notifications should be created via `create_notification` which returns zero-ed out memory anyway.

Reply:
Yeah that stuff about `notification_init`. I only moved these values to the top of the notification to show, that these are actually not really processed in `notification_init`.